kyo-scheduler-pekko/jvm/src/test/scala/kyo/scheduler/KyoExecutorServiceConfiguratorTest.scala

```diff
@@ -0,0 +1,105 @@
+package kyo.scheduler
+
+import com.typesafe.config.ConfigFactory
+import java.util.concurrent.CountDownLatch
+import org.apache.pekko.actor.Actor
+import org.apache.pekko.actor.ActorSystem
+import org.apache.pekko.actor.Props
+import org.apache.pekko.testkit.TestKit
+import org.apache.pekko.testkit.TestProbe
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.NonImplicitAssertions
+import org.scalatest.freespec.AnyFreeSpecLike
+import scala.concurrent.Promise
+import scala.concurrent.duration.*
+
+class KyoExecutorServiceConfiguratorTest
+    extends TestKit(ActorSystem(
+        "KyoTest",
+        ConfigFactory.parseString("""
+            pekko.actor.default-dispatcher {
+                type = "Dispatcher"
+                executor = "kyo.scheduler.KyoExecutorServiceConfigurator"
+            }
+        """)
+    ))
+    with AnyFreeSpecLike
+    with NonImplicitAssertions
+    with BeforeAndAfterAll {
+
+    override def afterAll(): Unit = {
+        TestKit.shutdownActorSystem(system)
+    }
+
+    val dispatcher = system.dispatchers.defaultGlobalDispatcher
+
+    case class TestException() extends Exception
+
+    "executes tasks on kyo threads" in {
+        val thread                 = Thread.currentThread()
+        val latch                  = new CountDownLatch(1)
+        var executorThread: Thread = null
+
+        dispatcher.execute { () =>
+            executorThread = Thread.currentThread()
+            latch.countDown()
+        }
+
+        latch.await(5, SECONDS)
+        assert(thread ne executorThread)
+        assert(executorThread.getName().contains("kyo"))
+    }
+
+    "handles messages through actors" in {
+        val latch                  = new CountDownLatch(1)
+        var executorThread: Thread = null
+
+        val probe = TestProbe()
+        val actor = system.actorOf(Props(new Actor {
+            def receive = {
+                case _ =>
+                    executorThread = Thread.currentThread()
```

Comment:
better with an `ask/?` and reply with the thread name,eg : https://github.com/apache/pekko/pull/1728

Reply:
updated